src/decorations/intervals.jl

```diff
@@ -29,44 +29,51 @@ of a previously executed sequence of functions acting on an initial interval.
 struct DecoratedInterval{T}
     interval::Interval{T}
     decoration::DECORATION
+
+    function DecoratedInterval{T}(I::Interval, d::DECORATION) where T
```

Comment:
A small note in the docstring to tell that the decoration is always checked at creation would be good.